### PR TITLE
Fix linting errors across the app

### LINKED ISSUE #434: Fix linting errors across the app
### Is your proposal related to a problem?
Even though linting rules are present, currently these aren't enforced using a pipeline to be strict about it.

### Describe the solution you'd like
Let's fix the lint errors and also add recommended/custom rules (like 80 line width) we'd want to have initially. After this is done, we can incorporate strict checking with a pipeline later on.

### Describe alternatives you've considered
None

### If the feature is approved, would you be willing to submit a PR?
Yes

### PR DESCRIPTION
**Pull request by arpitnath**
Fixes #434 
Fixes linting errors across the app and recommended vscode extensions and eslint settings.

### CODE REVIEW

File: .vscode/extension.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "recommendations": [
+      "dbaeumer.vscode-eslint",
```
Comment (gondar00) [inline .vscode/extension.json:3]: `formulahendry.auto-rename-tag` 
`xabikos.javascriptsnippets`
`streetsidesoftware.code-spell-checker`

mentioning some more extensions that I use generally

Comment (arpitnath) [reply .vscode/extension.json:3]: okay cool 👍🏻

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+    // ...settings,
```
Comment (gondar00) [inline .vscode/settings.json:2]: ❌

File: frontend/.eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    ],
+    "parser": "babel-eslint",
+    "parserOptions": {
+      "ecmaFeatures": {
+        "jsx": true
+      },
+      "ecmaVersion": 12,
+      "sourceType": "module"
+    },
+    "plugins": ["react", "prettier"],
```
Comment (gondar00) [inline frontend/.eslintrc.json:24]: If adding prettier here, probably can add it to the vscode extension suggestion

File: package.json [hunk 4]
```diff
@@ -12,6 +15,7 @@
         "@babel/preset-react": "^7.0.0",
         "babel-loader": "^8.0.5",
         "html-webpack-plugin": "^3.2.0",
+        "husky": "^7.0.2",
```
Comment (gondar00) [inline package.json:18]: Probably need to add 

```
"husky": {
  "hooks": {
    "pre-commit": "lint-staged"
  }
}
```

File: frontend/.eslintrc.json (hunk 3 above)
Comment (arpitnath) [reply frontend/.eslintrc.json:24]: but we are formating with eslint also I have disabled prettier. so should I add it?

File: frontend/cypress/plugins/index.js [hunk 5]
```diff
@@ -1,16 +1,14 @@
-const webpackPreprocessor = require('@cypress/webpack-preprocessor')
+// const webpackPreprocessor = require('@cypress/webpack-preprocessor');
```
Comment (akshaysasidrn) [inline frontend/cypress/plugins/index.js:1]: Can we remove the comments?

**Review by akshaysasidrn — CHANGES_REQUESTED** [review]

File: frontend/cypress/plugins/index.js (hunk 5 above)
Comment (arpitnath) [reply frontend/cypress/plugins/index.js:1]: yeah sure

File: server/package.json [hunk 6]
```diff
@@ -8,12 +8,11 @@
   "scripts": {
     "prebuild": "rimraf dist",
     "build": "nest build",
-    "format": "prettier --write \"src/**/*.ts\" \"test/**/*.ts\"",
+    "lint": "eslint . --fix '**/*.ts'",
```
[the remark this answers is not in the document]
Comment (arpitnath) [reply server/package.json:11]: also will separate in frontend too

**Review by akshaysasidrn — APPROVED** [review]

**Review by gondar00 — DISMISSED** [review]

**Pull request merged**